packages/connection-form/src/components/advanced-options-tabs/authentication-tab/authentication-gssapi.tsx

```diff
@@ -30,13 +44,22 @@ function AuthenticationGSSAPI({
     'kerberosPrincipal'
   );
   const principal = getConnectionStringUsername(connectionStringUrl);
+  const password = getConnectionStringPassword(connectionStringUrl);
+
   const authMechanismProperties =
     parseAuthMechanismProperties(connectionStringUrl);
   const serviceName = authMechanismProperties.get('SERVICE_NAME');
   const serviceRealm = authMechanismProperties.get('SERVICE_REALM');
-  const canonicalizeHostname = authMechanismProperties.get(
-    'CANONICALIZE_HOST_NAME'
-  );
+  const canonicalizeHostname =
+    authMechanismProperties.get('CANONICALIZE_HOST_NAME') || 'none';
+
+  const [showPassword, setShowPassword] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (!showPassword && password.length) {
+      setShowPassword(true);
+    }
+  }, [password, showPassword, updateConnectionFormField]);
```

Comment:
I have very limited understanding of how these things work in react, but should there also be a reverse transition for `setShowPassword(false);` when there is no password? I just tried this locally, if I add a password in the connection string, the `Provide password directly` box becomes checked and the password filled in, but if I remove the password in the connection string the box doesn’t become unchecked (this is a pretty minor thing of course, I’m mostly asking because this UI feature seems to be a good template for the TLS system CA option)